### PR TITLE
Refactor: Use `jest` testing library for running unit tests

### PR DESCRIPTION
**Pull request by aloisklink**
## :bookmark_tabs: Summary

Use the [jest](https://github.com/facebook/jest) JavaScript testing library to run unit tests, as discussed in https://github.com/mermaid-js/mermaid-cli/pull/322#issuecomment-1168392106

While I was at it, I also added:
- a GitHub Actions CI test for running `yarn test` on NodeJS 14, 16, 18
- tested the contents of the error messages for:
  - mmdc failing due to puppeteer timeout error
  - mmdc failing due to mermaid syntax error
- added `pdf` as an output format, so now `.mmd` files are tested for `png`, `svg`, `pdf`
- tested all `*.md` files, instead of just `markdown-output.md`
  `.md` files are now tested for `png`, `svg`, `pdf`, and `md`

This is what the tests look like when running `yarn test`

image

And if I make an error, this is what failing tests look like:

image

## :straight_ruler: Design Decisions

I removed the logging of `stdout` on tests, to simplify the test output.

Also, I've made all the `workflows/test-positive` and `workflows/test-negative` run concurrently. This means that unit tests is a lot faster (especially when you have a lot of CPU cores), but it does take more RAM (about 1 GB on my computer).

## :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid-cli/blob/master/CONTRIBUTING.md)
  - Again, I've ignored the `standard` styling to make the `git diff` as simple as possible, but I made sure everything was autofixable with `npx standard --fix`.
- [x] :computer: have added unit/e2e tests (if appropriate)
- [x] :bookmark: targeted `master` branch

### CODE REVIEW

File: src-test/test.js [hunk 1]
```diff
@@ -1,54 +1,49 @@
...
-    } catch (err) {
-      console.warn(`${file}: child process failed with error: ${err.message}`);
-
-      reject(err);
-    }
-  });
+  const result = file.replace(/\.(?:mmd|md)$/, "-stdin." + format);
+  // exec will throw with stderr if there is a non-zero exit code
+  return await promisify(exec)(`cat ${workflow}/${file} | \
+    node index.bundle.js -o ${out}/${result} -c ${workflow}/config.json`
```
[the remark this answers is not in the document]
Comment (aloisklink) [reply src-test/test.js:30]: This shouldn't be too much of an issue. It does mean that somebody could make a test file called `workflows/" ; echo 'shell injection'; .md"` to do a shell injection attack, but we'd be able to see it before running the code, so it should be okay.

Ideally, we could change this to something like the following (e.g. using NodeJS to pipe the file to stdin):

```js
  const promise = promisify(execFile)('node', [
    'index.bundle.js',
    '-o', join(out, result),
    '-c', join(workflow, 'config.json'),
  ], {
    stdio: "null",
  })
  const inputFileStream = createReadStream(join(workflow, file))
  inputFileStream.pipe(promise.child.stdin)

  await promise
```

However, there seems to be a bug in `index.js`. fs.fstatSync(0) can also be a `isSocket()` when `stdin` is being piped from NodeJS, not bash. However, it's also `isSocket()` when `stdin` is empty, so it won't be an easy fix.

https://github.com/mermaid-js/mermaid-cli/blob/4a86a85624a1dfa3b22bb9c0a165905275c90248/src/index.js#L26
